### ISSUE TITLE
Bounded TypeVar's attributes not accessable when defined for a variable

### ISSUE THREAD

**Issue by dumbPy** [microsoft/pyright#427]
**Describe the bug**
If bounded TypeVar is defined explicitly for a variable, it's inner attributes of the bound are not accessed.
**To Reproduce**
Call method that returns `Any` and define type explicitly for the variable (code below)

**Expected behavior**
Should read inner attributes (clear in the code below)

**Screenshots or Code**

```python
from typing import TypeVar, Any

class Parent:
    prop = None

# type hint for any object of subclass of Parent, right?
AnyChildOfParent = TypeVar("AnyChildOfParent", bound=Parent)

class Child(Parent):
    pass

def func_with_inferable_return_type(): return Child()
def func_without_inferable_return_type()->Any: pass

X = func_with_inferable_return_type()
X.prop = 1 # works as the type of X in infered

Y:AnyChildOfParent = func_without_inferable_return_type()
Y.prop = 1 # throws error, 
# Cannot assign member 'prop' for type 'TypeVar['AnyChildOfParent']'
#   Unsupported type 'TypeVar['AnyChildOfParent']'

def some_func_that_takes_child(x:AnyChildOfParent)->None:
    x.prop = 1 # the same works when type is defined in function argument

# The same works for native types at lease (others not checked)
Z:str = func_without_inferable_return_type()
Z.split() # does not throw an error 
```

**VS Code extension or command-line**
pyright extension v1.1.10

**Additional context**
More example in continuation of the above code (more closer to my usecase)
```python
# some module I imported that contained the above classes
some_module = ["Parent", "Child"]

# Asume I imported AnyChildOfParent Exclusively and the rest imports are based
# on some json config hence imported using getattr and string key form json

# Though getattribute might return help in typechecking, I exclusively define it
y:AnyChildOfParent = some_module.__getattribute__("Child")
y.prop = None
# Cannot assign member 'prop' for type 'TypeVar['AnyChildOfParent']'
# Unsupported type 'TypeVar['AnyChildOfParent']
```

**Comment by erictraut** [comment]
Thanks for the bug report. This issue is already addressed by a set of changes that I just checked in. It will be fixed in the next release of pyright.

**Comment by dumbPy** [comment]
Thankyou !!!

**Comment by dumbPy** [comment]
ps, pyright is damn awesome !! :smile:
